Source/WebCore/accessibility/cocoa/AXTextMarkerCocoa.mm

```diff
@@ -91,7 +92,40 @@
         , m_end.platformData().autorelease()
     ));
 }
-#endif // PLATFORM(MAC)
+
+#elif PLATFORM(IOS_FAMILY)
+
+AXTextMarkerRange::AXTextMarkerRange(NSArray *markers)
+{
+    if (markers.count != 2)
+        return;
+
+    WebAccessibilityTextMarker *start = [markers objectAtIndex:0];
+    WebAccessibilityTextMarker *end = [markers objectAtIndex:1];
+    if (![start isKindOfClass:[WebAccessibilityTextMarker class]] || ![end isKindOfClass:[WebAccessibilityTextMarker class]])
+        return;
+
+    m_start = { [start textMarkerData ] };
+    m_end = { [end textMarkerData] };
+}
+
+NSArray *AXTextMarkerRange::platformData() const
+{
+    if (!*this)
+        return nil;
+
+    RefPtr object = m_start.object();
+    ASSERT(object); // Since *this is not null.
+    auto* cache = object->axObjectCache();
+    if (!cache)
+        return nil;
+    return @[
+        adoptNS([[WebAccessibilityTextMarker alloc] initWithTextMarker:&m_start.m_data cache:cache]).autorelease(),
+        adoptNS([[WebAccessibilityTextMarker alloc] initWithTextMarker:&m_end.m_data cache:cache]).autorelease()
+    ];
```

Comment:
Does the returned array need to be autoreleased as well?

Reply:
Yes, I should probably return a RetainPtr<NSArray> here, thanks for catching.